scanner/src/main/kotlin/LocalScanner.kt

```diff
@@ -177,19 +247,8 @@ abstract class LocalScanner(name: String, config: ScannerConfiguration) : Scanne
      * The return value is a list of [ScanResult]s. If a package could not be scanned, a [ScanException] is thrown.
      */
     private fun scanPackage(scannerDetails: ScannerDetails, pkg: Package, outputDirectory: File,
-                    downloadDirectory: File): List<ScanResult> {
-        val scanResultsForPackageDirectory = File(outputDirectory, pkg.id.toPath()).apply { safeMkdirs() }
-        val resultsFile = File(scanResultsForPackageDirectory, "scan-results_${scannerDetails.name}.$resultFileExt")
-
-        val storedResults = ScanResultsStorage.read(pkg, scannerDetails)
-
-        if (storedResults.results.isNotEmpty()) {
-            // Some external tools rely on the raw results filer to be written to the scan results directory, so write
-            // the first stored result to resultsFile. This feature will be removed when the reporter tool becomes
-            // available.
-            resultsFile.mapper().writeValue(resultsFile, storedResults.results.first().rawResult)
-            return storedResults.results
-        }
+                            downloadDirectory: File): List<ScanResult> {
+        val resultsFile = getResultsFile(scannerDetails, pkg, outputDirectory)
```

Comment:
Would it be possible to do the split of `scanPackage()` into `getResultsFile()` and `readFromCache()` already in a previous refactoring commit to reduce the diff when introducing coroutines?

Reply:
Done.